software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+      set the visible of button "snapCaptureButt" stack "NodeDetails" to true
+      if tDeets[pLine]["figure"] is not "[empty]" or (tDeets[pLine]["new"] is not empty and tDeets[pLine]["new"] is not "hide") then
+         -- allow clear if 1) [figure] already has a file or 2) a new file has been loaded in [new]
+         -- don't allow clear if "hide" is in [new], which is trying to get rid of what's in [figure]
+         set the visible of button "clearImgButt" stack "NodeDetails" to true 
+         set the visible of button "expandImgButt" stack "NodeDetails" to true 
+      end if
+   else if tLabel is among the words "title memo weight" then
+      set the visible of field "editData" stack "NodeDetails" to true
+      set the visible of field "tagLabel" stack "NodeDetails" to true
+      set the visible of image "figureHolder" stack "NodeDetails" to false
+      set the visible of button "loadImgButt" stack "NodeDetails" to false
+      set the visible of button "clearImgButt" stack "NodeDetails" to false 
+      set the visible of button "expandImgButt" stack "NodeDetails" to false 
+      set the visible of button "setCaptureButt" stack "NodeDetails" to false
+      set the visible of button "snapCaptureButt" stack "NodeDetails" to false
+      put tLabel into field "tagLabel" stack "NodeDetails"
+      if tDeets[pLine]["new"] is not empty and tDeets[pLine]["new"] is not "hide" then
+         put tDeets[pLine]["new"] into field "editData" stack "NodeDetails"
+      else 
+         put tDeets[pLine][tLabel] into field "editData" stack "NodeDetails"
+      end if
+      select the text of field "editData" stack "NodeDetails"
+   end if -- if label is divider
+   set the cNodeDetails of stack "NodeDetails" to tDeets
+end detailsListMouseUp
+
+on loadImgButtMouseUp pLine
+   -- on nodeDetails for loading a figure from an existing file
+   put the cNodeDetails of stack "NodeDetails" into tDeets
+   put specialfolderpath("documents") into tStart
+   answer file "choose an image (it doesn't actually force you to pick an image file at the moment)" with tStart
+   if the result is not cancel then
+      put empty into tDeets[pLine]["new"]
+      put true into tDeets["newFigure"]
+      set the itemdelimiter to slash
+      put it into tImgPath
+      put item -1 of tImgPath into tImgName
+      if sMain[sNowDoc][0]["filepath"] is not empty then
+         put sMain[sNowDoc][0]["filepath"] into tProjFolder
+         delete item -1 of tProjFolder
+         set the defaultfolder to tProjFolder
+         put the files into tFiles
+         if tImgName is among the lines of tFiles then
+            ask "This filename is already in use. If you don't change it the old file will be overwritten. (Include the file extension)" with tImgName
+            put it into tDeets[pLine]["new"]["newName"]
+         end if
+      end if -- the project has a save location
+      put tImgPath into tDeets[pLine]["new"]["copyFrom"]
+      -- put the image on screen
```

Comment:
@matthewmaier I think some extra vertical whitespace in your code would aid readability. I've commented here because this is probably a good example where just adding an extra line above the comment would help to visually break up the code.